source/plugin/data/custom/datamanipulators.rst

```diff
@@ -91,15 +93,142 @@ the work for you. In addition to the above, you need to override the following m
 
 - ``getValueGetter()`` should pass the :doc:`Value <values>` representing your data. It is recommented that this is 
   cached in the immutable version of your data.
-- ``fill(DataHolder, MergeFunction)`` should replace the data on your object with that of the given ``DataHolder``, using
-  the result of ``MergeFunction#merge()``. See the implementation below for an example.
+- ``fill(DataHolder, MergeFunction)`` should replace the data on your object with that of the given ``DataHolder``, 
+  using the result of ``MergeFunction#merge()``. See the implementation below for an example.
 - ``from(DataContainer)`` should overwrite its value with the one in the container and return itself, otherwise return
   ``Optional.empty()``
 - ``copy()`` should, as the name, return a copy of itself with the same data.
 
-.. TODO: Custom single type impl
+**Code Example: Implementation of a Custom Single Data**
```

Comment:
Please note that comments in code blocks _don't get translated_ (code blocks bypass translation). If you can find places for the comments in the body text, that would be preferable (but not always possible). This is mentioned in the SpongeDocs style guide (under contributing), so check it out if you're not familiar with the guidelines.